packages/suite/src/actions/suite/initAction.ts

```diff
@@ -0,0 +1,59 @@
+import { SUITE } from './constants';
+
+import * as suiteActions from '@suite-actions/suiteActions';
+import * as routerActions from '@suite-actions/routerActions';
+import * as analyticsActions from '@suite-actions/analyticsActions';
+import * as messageSystemActions from '@suite-actions/messageSystemActions';
+import * as languageActions from '@settings-actions/languageActions';
+import * as trezorConnectActions from '@suite-actions/trezorConnectActions';
+import * as blockchainActions from '@wallet-actions/blockchainActions';
+
+import type { Dispatch, GetState } from '@suite-types';
+
+export const init = () => async (dispatch: Dispatch, getState: GetState) => {
+    const {
+        suite: {
+            settings: { language },
+            lifecycle: { status },
+        },
+        analytics,
+    } = getState();
+
+    if (status !== 'initial') return;
+
+    dispatch({ type: SUITE.INIT });
+
+    dispatch(suiteActions.initDevices());
+
+    // right after storage is loaded, we might start:
+
+    // 1. init analytics
+    dispatch(analyticsActions.init(analytics));
+
+    // 2. fetching locales
+    dispatch(languageActions.setLanguage(language));
+
+    // 3. fetch message system config
+    await dispatch(messageSystemActions.init()).catch(err => console.error(err));
+
+    // 4. redirecting user into welcome screen (if needed)
+    dispatch(routerActions.initialRedirection());
+
+    // 5. init connect (could throw an error,
+    // then the error is caught in <ErrorBoundary /> in Main.tsx
+    try {
+        await dispatch(trezorConnectActions.init());
+    } catch (err) {
+        dispatch({ type: SUITE.ERROR, error: err.message });
+        throw err;
+    }
```

Comment:
If I understand well, this is the reason we do not need `CONNECT_INITIALIZED` anymore? because it will be initialized or it will throw error and will not be initialized.
So we can assume that it is initialized if there was no error?

Is that assumption right or am I missing something?

Reply:
Exactly. Because previously, the subsequent steps were performed by middleware on `SUITE.CONNECT_INITIALIZED` (which was dispatched by `trezorConnectActions.init` itself) and if it failed, `SUITE.ERROR` was dispatched instead (which is currently the same, except it's from `init` instead of `trezorConnect.init`).